README.md

```diff
@@ -27,68 +27,225 @@ Installation
+        'collectors.console' => [
+            \App\Debug\AwesomeConsoleCollector::class,
+        ],
+    ],
+];
+```
+
+Under `yiisoft/yii-debug` configuration you may use:
+1. `collectors` key for both web and console runs
+2. `collectors.web` key only for web session
+3. `collectors.console` key only for console run
+
+> Do not register a collector for a session where the collector will not collect any payload.
+
+
+The lines above connect collectors with a debug extension run. 
+Under the hood it just calls `getCollected()` method from the collectors at the end of application cycle run.
+
+### Event listener collector
+
+Subscribe to any events you want with adding a listener into the configuration:
+
+With [`yiisoft/event-dispatcher`](https://github.com/yiisoft/event-dispatcher) configuration it may look like the following:
+
+```php
+return [
+    \Yiisoft\Yii\Http\Event\BeforeRequest::class => [
+        [\App\Debug\AwesomeCollector::class, 'collect'],
+    ],
+];
+```
+
+Each `Yiisoft\Yii\Http\Event\BeforeRequest` triggered event will call `App\Debug\AwesomeCollector::collect($event)` method, 
+so you can collect any data from the event or call any other services to enrich the event data with additional payload.
+
+### Proxy collector
+
+Proxy collectors are used in case you want to decorate a service from DIC and sniff methods' calls with its values.
+
+First you need to create a class that will work as a decorator. See https://en.wikipedia.org/wiki/Decorator_pattern if you are new with it.
+
+Decorators may inject any services through `__construct` method, but you should specify services you like to wrap.
+Into section `trackedServices` of `yiisoft/yii-debug` configuration you should specify:
+1. A service you want to decorate
+2. A decorator that will decorate the service
+3. A collector that will be injected into the decorator
+
+Syntax of configuration is: `Decorating service => [Decorator, Collector]`. 
+
+Despite adding the tracking service configuration you still need to register the collector if you did not do it before.
+Whole configuration of added proxy collector looks like the following:
+
+```php
+return [
+    'yiisoft/yii-debug' => [
+        'collectors' => [
+            \App\Debug\AwesomeCollector::class,
+        ],
+        'trackedServices' => [
+            // Decorating service => [Decorator, Collector],
+            \Psr\Log\LoggerInterface::class => [\App\Debug\LoggerInterfaceProxy::class, \App\Debug\AwesomeCollector::class],
         ],
-        // ...
     ],
-    ...
 ];
 ```
 
-You must make some changes to your OS. See these examples: 
- - PHPStorm: https://github.com/aik099/PhpStormProtocol
- - Sublime Text 3 on Windows or Linux: https://packagecontrol.io/packages/subl%20protocol
- - Sublime Text 3 on Mac: https://github.com/inopinatus/sublime_url
+#### Index collector
 
-#### Virtualized or dockerized
+Index collector is a collector that provides additional "summary" payload. 
```

Comment:
Is it a good idea to rename it from "index" collector to "summary" collector?

Reply:
Sounds good

Reply:
https://github.com/yiisoft/yii-debug/pull/210/commits/3efc7bbf5a5ab0915d63f8ed0f0f61cf8c23ece5